dask/dataframe/io/parquet/arrow.py

```diff
@@ -807,9 +807,15 @@ def _collect_dataset_info(
                     gather_statistics = True
             elif require_extension:
                 # Need to materialize all paths if we are missing the _metadata file
+                # Raise error if all files have been filtered by extension
+                len0 = len(paths)
                 paths = [
                     path for path in fs.find(paths) if path.endswith(require_extension)
                 ]
+                if len0 and paths == []:
+                    raise ValueError(
+                        "No files satisfy the `required_extension` criteria."
```

Comment:
I think this error would be slightly more informative if it included the value of `require_extension`
```suggestion
                        "No files satisfy the `required_extension` criteria "
                        f"(files must end with {required_extension})."
```